### ISSUE TITLE
[Kernel 4.4] Kernelsu freeze on installing modules with KSU app

### ISSUE THREAD

**Issue by Aryanxfa** [tiann/KernelSU#1050]
**Describe the bug**
Any module installation gets stuck at "Mounting Image" and the device freezes. Need to perform a force reboot to restore functionality. The thing to note is that turning selinux permissive allows for proper installation of modules. 

**To Reproduce**
Installing any module causes that issue. 

**Expected behaviour**
The module installation should either give a clear failure or a success without freezing the device

**Screenshots**
![photo_6154388858008091188_y](https://github.com/tiann/KernelSU/assets/26765117/e3cc89f9-a84b-4bf1-af7a-130547a7458f)

**Smartphone (please complete the following information):**
 - Device: Samsung Galaxy A20 ( same on a10,a20e,a30,a30s,a40 )
 - OS: Cherish Android 13 v4.12 ( tested with lineage 20.0, same issue )

**Additional context**
As the problem is solved after setting selinux to permissive. i have pulled some ksu app logs, module installation logs and avc_msg
attaching them here for your reference. i am astonished as to why u:r:su:s0 is getting denials.

`type=1400 audit(1695268737.027:3064): avc:  denied  { ioctl } for  pid=19800 comm="e2fsck" path="/data/adb/ksu/update_tmp.img" dev="mmcblk0p32" ino=785282 ioctlcmd=127c scontext=u:r:su:s0 tcontext=u:object_r:adb_data_file:s0 tclass=file permissive=1
`

Kernel source:- https://github.com/eurekadevelopment/Eureka-Kernel-Exynos7885-Q-R-S
KernelSu commit :- https://github.com/eurekadevelopment/Eureka-Kernel-Exynos7885-Q-R-S/commit/41d4196ec2fe6543893b378abe75dba4eb5ceef9
Device tree:- https://github.com/eurekadevelopment/android_device_samsung_exynos7885/

[KernelSU_bugreport_2023-10-16_22_41.tar.gz](https://github.com/tiann/KernelSU/files/12918933/KernelSU_bugreport_2023-10-16_22_41.tar.gz)

[KernelSU_install_log_2023-10-16-22-40-09.log](https://github.com/tiann/KernelSU/files/12918938/KernelSU_install_log_2023-10-16-22-40-09.log)

[avc_msg.txt](https://github.com/tiann/KernelSU/files/12919001/avc_msg.txt)

Any suggestion / guidance in the right direction would be extremely helpful

Thanks!

**Comment by tiann** [comment]
try add `ksu_allowxperm(db, KERNEL_SU_DOMAIN, ALL, "file", ALL);` [here](https://github.com/tiann/KernelSU/blob/main/kernel/selinux/rules.c#L65)

**Comment by Aryanxfa** [comment]
> try add `ksu_allowxperm(db, KERNEL_SU_DOMAIN, ALL, "file", ALL);` [here](https://github.com/tiann/KernelSU/blob/main/kernel/selinux/rules.c#L65)

Hi tiann, Thanks for the reply!
I tried your solution but the same problem still exists with enforcing.
I can still install modules with permissive. 
This time avc no longer shows any issue with u:r:su:s0 instead with kdevtmpfs. i am attaching the same for your reference.
```
type=1400 audit(1697558763.343:261): avc:  denied  { create } for  pid=63 comm="kdevtmpfs" name="loop38" scontext=u:r:kernel:s0 tcontext=u:object_r:device:s0 tclass=blk_file permissive=1
type=1400 audit(1697558763.343:262): avc:  denied  { setattr } for  pid=63 comm="kdevtmpfs" name="loop38" dev="devtmpfs" ino=50486 scontext=u:r:kernel:s0 tcontext=u:object_r:device:s0 tclass=blk_file permissive=1

```
[avc_msg.txt](https://github.com/tiann/KernelSU/files/12936512/avc_msg.txt)

**Comment by tiann** [comment]
You can add these two rules and try again.

**Comment by Aryanxfa** [comment]
Hi tiann, 
It is now working correctly. 
Thanks for the solution. I appreciate your time.
you may close this issue. Issue has been resolved

**Comment by TogoFire** [comment]
> try add `ksu_allowxperm(db, KERNEL_SU_DOMAIN, ALL, "file", ALL);` [here](https://github.com/tiann/KernelSU/blob/main/kernel/selinux/rules.c#L65)

This helped on my 4.9 kernel. But the module is still not installed and when trying to install any module the device restarts.
I suspect it's something in defconfig that's getting in the way, but I still don't know exactly what it is.

I have these commits due to broken kprobes:
security: selinux: allow init exec (kernelsu:ksud) under nosuid
drivers: input: Add KernelSU SafeMode inpnt_event Kernel Hook
fs: Add hook kernel hook for kernelsu (exec, open, read_write, stat)

The modules were supposed to remain ok, even without kprobes and after the commits mentioned above, but that doesn't work for me. I tried with kernelsu in gitmodule and manually. And I checked selinux.

git: https://github.com/Dev-msm8953/kernel_xiaomi_msm8953
